File: android/api-tests/src/test/java/com/revenuecat/apitests/java/OnResultListApiTests.java
package com.revenuecat.apitests.java;

import com.revenuecat.purchases.hybridcommon.ErrorContainer;
import com.revenuecat.purchases.hybridcommon.OnResultList;

import java.util.List;
import java.util.Map;

@SuppressWarnings("unused")
class OnResultListApiTests {
    private void checkOnSuccess(OnResultList onResult,
                                List<Map<String, ?>> result) {
        onResult.onReceived(result);
    }

    private void checkOnError(OnResultList onResult,
                              ErrorContainer errorContainer) {
        onResult.onError(errorContainer);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/OnResultListApiTests.kt
package com.revenuecat.apitests.kotlin

import com.revenuecat.purchases.hybridcommon.ErrorContainer
import com.revenuecat.purchases.hybridcommon.OnResultList

@Suppress("unused")
class OnResultListApiTests {
    fun checkOnReceived(
        onResult: OnResultList,
        resultMap: List<Map<String, Any>>,
    ) {
        onResult.onReceived(resultMap)
    }

    fun checkOnError(
        onResult: OnResultList,
        errorContainer: ErrorContainer,
    ) {
        onResult.onError(errorContainer)
    }
}
